src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java

```diff
@@ -802,11 +809,20 @@ private ModelBuildingRequest createModelBuildingRequest(File pomFile) {
                 context,
                 project.getRemoteProjectRepositories(),
                 getReactorModelsFromSession());
-        Properties userProperties = this.session.getUserProperties();
-        List<String> activeProfiles = this.session.getRequest().getActiveProfiles();
+        Properties userAndActiveExternalProfilesProperties = new Properties();
+        userAndActiveExternalProfilesProperties.putAll(this.session.getUserProperties());
+        settings.getProfiles().stream()
+                .filter(p -> settings.getActiveProfiles().contains(p.getId()))
+                .forEach(
+                        activeProfile -> userAndActiveExternalProfilesProperties.putAll(activeProfile.getProperties()));
```

Comment:
We also have `session.getRequest().getInactiveProfiles()` 
https://maven.apache.org/guides/introduction/introduction-to-profiles.html#deactivating-a-profile

so profile id should be on ActiveProfiles from session request or from settings and not in InactiveProfiles list when build properties list

Reply:
Good catch, done 👍